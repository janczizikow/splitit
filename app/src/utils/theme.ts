const colors = {
  primary: "#00ccbb",
  error: "#ff483b",
  errorBg: "#ffebe4",
  success: "#006d68",
  successBg: "#edfad0",
  border: "#bac3c3",
  greyBg: "#f9fafa",
  lightGreyBg: "#f5f5f5",
  headingColor: "#2e3333"
};

const borderRadius = 2;

export interface Theme {
  colors: typeof colors;
  borderRadius: number;
}
export default {
  colors,
  borderRadius
};
