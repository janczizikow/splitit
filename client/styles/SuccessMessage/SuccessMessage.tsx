import styled from "@emotion/styled";

const SuccessMessage = styled.div`
  padding: 1rem;
  margin-bottom: 1.5rem;
  font-size: 0.875rem;
  color: ${p => p.theme.colors.success};
  text-align: center;
  background-color: ${p => p.theme.colors.successBg};
`;

SuccessMessage.displayName = "SuccessMessage";

export default SuccessMessage;
