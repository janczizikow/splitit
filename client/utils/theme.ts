const colors = {
  primary: "#00ccbb",
  error: "#ff483b",
  errorBg: "#ffebe4",
  success: "#006d68",
  successBg: "#edfad0",
  border: "#bac3c3",
  greyBg: "#f9fafa",
  lightGreyBg: "#f5f5f5",
  headingColor: "#2e3333"
};

const borderRadius = 2;

const gridGutters = 16;
const gridColumns = 12;
const gridBreakpoints = [0, 576, 768, 992, 1200];

export interface Theme {
  colors: typeof colors;
  borderRadius: number;
  gridGutters: number | string;
  gridColumns: number;
  gridBreakpoints: number[];
}

export default {
  colors,
  borderRadius,
  gridGutters,
  gridColumns,
  gridBreakpoints
};
