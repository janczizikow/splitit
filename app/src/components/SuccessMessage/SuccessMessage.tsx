import React from "react";
import { View, StyleSheet } from "react-native";
import { Text } from "react-native-elements";
import theme from "../../utils/theme";

interface Props {
  children?: string;
}

const SuccessMessage: React.FunctionComponent<Props> = ({ children }) => (
  <View style={styles.container}>
    <Text style={styles.successText}>{children}</Text>
  </View>
);

const styles = StyleSheet.create({
  container: {
    marginVertical: 24,
    marginHorizontal: 12,
    padding: 16,
    textAlign: "center",
    backgroundColor: theme.colors.successBg
  },
  successText: {
    color: theme.colors.success
  }
});

export default SuccessMessage;
