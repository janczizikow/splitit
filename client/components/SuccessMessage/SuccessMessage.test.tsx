import { shallow, mount } from "enzyme";
import toJson from "enzyme-to-json";
import SuccessMessage from "./SuccessMessage";
import theme from "../../utils/theme";

describe("<SuccessMessage/>", () => {
  it("renders without crashing", () => {
    const wrapper = shallow(<SuccessMessage />);
    expect(wrapper).toHaveLength(1);
  });

  it("matches snapshot", () => {
    const wrapper = mount(<SuccessMessage theme={theme} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
