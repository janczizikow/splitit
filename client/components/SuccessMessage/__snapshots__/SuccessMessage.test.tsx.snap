// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SuccessMessage/> matches snapshot 1`] = `
.emotion-0 {
  padding: 1rem;
  margin-bottom: 1.5rem;
  font-size: 0.875rem;
  color: #006d68;
  text-align: center;
  background-color: #edfad0;
}

<SuccessMessage
  theme={
    Object {
      "borderRadius": 2,
      "colors": Object {
        "border": "#bac3c3",
        "error": "#ff483b",
        "primary": "#00ccbb",
        "success": "#006d68",
        "successBg": "#edfad0",
      },
      "gridBreakpoints": Array [
        0,
        576,
        768,
        992,
        1200,
      ],
      "gridColumns": 12,
      "gridGutters": 16,
    }
  }
>
  <div
    className="emotion-0"
  />
</SuccessMessage>
`;
